### PR TITLE
fix: take srcDir and outDir from docgen.json config file into account

### PR DESCRIPTION
**Pull request by parischap**
## Type

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related

- Related Issue #
- Closes #

### CODE REVIEW

File: src/Configuration.ts [hunk 1]
```diff
@@ -285,8 +285,20 @@ export const load = (args: {
...
 
+    const srcDir = Option.match(config, {
+      onNone: () => args.srcDir,
+      onSome: ({ srcDir }) => srcDir || args.srcDir,
+    })
+
+    const outDir = Option.match(config, {
+      onNone: () => args.outDir,
+      onSome: ({ outDir }) => outDir || args.outDir,
+    })
```
Comment (gcanti) [inline src/Configuration.ts:296]:
```suggestion
    const srcDir = config.pipe(
      Option.flatMapNullable((config) => config.srcDir),
      Option.getOrElse(() => args.srcDir)
    )

    const outDir = config.pipe(
      Option.flatMapNullable((config) => config.outDir),
      Option.getOrElse(() => args.outDir)
    )
```

Comment (parischap) [reply src/Configuration.ts:296]: Yes, your solution is more elegant.

Do you want me to modify my PR?

Comment (gcanti) [reply src/Configuration.ts:296]: (FYI if you agree with the changes you can simply click the "Commit suggestion" button)

Comment (parischap) [reply src/Configuration.ts:296]: Sorry I don t have access to my PC tonight. And no such button on the phone github version. Will do it tomorrow morning.

Thanks and good night

**Pull request merged**